fairscale/nn/data_parallel/fully_sharded_data_parallel.py

```diff
@@ -619,10 +619,15 @@ def clip_grad_norm_(
         local_norm = calc_grad_norm(params_with_grad, norm_type).cuda()
         if norm_type == inf:
             total_norm = local_norm
-            dist.all_reduce(total_norm, op=torch.distributed.ReduceOp.MAX, group=self.process_group)
+            dist.all_reduce(
```

Comment:
The all_reduce here is after both fwd and bwd passes are done, right? We need to pick the right PG and async flags for it. I think what you picked here (i.e. using the scatter reduce PG and use async=False) makes sense. But please use profile and dump and trace and let's review the trace on this.